src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.ConflictResolution.targets

```diff
@@ -0,0 +1,87 @@
+<!--
+***********************************************************************************************
+Microsoft.NET.ConflictResolution.targets
+
+WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and have
+          created a backup copy.  Incorrect changes to this file will make it
+          impossible to load or build your projects from the command-line or the IDE.
+
+Copyright (c) .NET Foundation. All rights reserved. 
+***********************************************************************************************
+-->
+
+<Project xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+
+  <PropertyGroup>
+    <MSBuildAllProjects>$(MSBuildAllProjects);$(MSBuildThisFileFullPath)</MSBuildAllProjects>
+  </PropertyGroup>
+
+    <!-- The condition on this target causes it to be skipped during design-time builds if
+        the restore operation hasn't run yet.  This is to avoid displaying an error in
+        the Visual Studio error list when a project is created before NuGet restore has
+        run and created the assets file. -->
+  <Target Name="_GetLockFileAssemblies"
+          DependsOnTargets="ComputePrivateAssetsPackageReferences;
+                            _DefaultMicrosoftNETPlatformLibrary"
+          Condition=" '$(DesignTimeBuild)' != 'true' Or Exists('$(ProjectAssetsFile)')">
+    <!-- Essentially a copy of the SDKs RunResolvePublishAssemblies.
```

Comment:
You may be able to remove this target if you can somehow make the SDK targets provide the right data.

The purpose of this target is to get all the runtime assemblies as items so that they can be trimmed from the deps file.  When I first implemented this, there wasn't a good item I could use that represented the runtime items at build time but that may have changed or I may not have seen it.

Runtime items are read for publish, since publish needs to do the copy.

I was thinking that if the entire deps content flowed through msbuild this target could certainly go away since the runtime items which eventually get written to the deps file for build would be present.  Barring that it seems like the SDK could at least populate a similar item.  That would really help to get rid of this hack.